addons/binding/org.openhab.binding.nest/ESH-INF/thing/structure.xml

```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nest" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="structure" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>Nest Structure</label>
+		<description>The details about the nest structures</description>
+
+		<channels>
+			<channel id="country_code" typeId="text">
+				<label>Country Code</label>
+				<description>The country code for the structure</description>
+			</channel>
+			<channel id="postal_code" typeId="text">
+				<label>Postal Code</label>
+				<description>The postal code of the structure</description>
+			</channel>
+			<channel id="peak_period_start_time" typeId="date">
+				<label>Peak period start</label>
+				<description>The start of the peak period.</description>
+			</channel>
+			<channel id="peak_period_end_time" typeId="date">
+				<label>Peak period end</label>
+				<description>The end of the peak period.</description>
+			</channel>
+			<channel id="time_zone" typeId="text">
+				<label>Time Zone</label>
+				<description>The time zone for the structure.</description>
+			</channel>
+			<channel id="eta_begin" typeId="date">
+				<label>ETA</label>
+				<description>The estimated time of arrival for someone to get home.</description>
+			</channel>
+			<channel id="co_alarm_state" typeId="alarmstate" >
+                <label>CO Alarm</label>
+                <description>The Carbon monoxide alarm state for the structure.</description>
+			</channel>
+            <channel id="smoke_alarm_state" typeId="alarmstate" >
+                <label>Smoke Alarm</label>
+                <description>The Smoke alarm state for the structure.</description>
+            </channel>
+			<channel id="rush_hour_rewards_enrollment" typeId="rushHour">
+				<label>Rush Hour Rewards</label>
+				<description>If rush hour rewards is turned on or off.</description>
+			</channel>
+			<channel id="away" typeId="away" />
+		</channels>
+
+		<properties>
+			<property name="vendor">Nest</property>
+		</properties>
+	</thing-type>
+
+	<channel-type id="away">
+		<item-type>String</item-type>
+		<label>Away</label>
+		<description>Away state of the structure</description>
+		<state>
+			<options>
+				<option value="away">Away</option>
+				<option value="home">Home</option>
+				<option value="autoaway">Auto Away</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="text">
+		<item-type>String</item-type>
+		<label>Country Code</label>
+		<description>Country code of the structure</description>
+	</channel-type>
+
+	<channel-type id="date">
```

Comment:
same comment as before about "String" and "Switch" - please change it to a semantic channel, where it is clear what the date expresses.